### PR TITLE
Quiz Refactored Animation 

### PR DESCRIPTION
**Pull request by LuckyHariharan**
I refactored the quiz animation to remove setTimeout. Instead, the animationComplete state variable is used to ensure the next stage only appears when the animation is complete. ```
({animationComplete &&...)}```

If this approach is reasonable, im happy to modify the other quizzes to pass the necessary props/set it up correctly.

Additionally, I:
- Removed the unnecessary arrow function
- Tested career and coding quizzes with no issues

Please let me know what you think of my approach @iamnotreddy

### CODE REVIEW

File: pages/resources/quizzes/careerQuiz/index.tsx [hunk 1]
```diff
@@ -123,21 +123,17 @@ const CareerQuiz = () => {
 
   const handleNextClick = () => {
     setTriggerAnimation(false);
-    handleQuizResponseMutations(quizViewState);
-    setTimeout(() => {
-      setTriggerAnimation(true);
-
```
Comment (LuckyHariharan) [inline pages/resources/quizzes/careerQuiz/index.tsx:129]: removed setTimeout fron handlenext and handlebackclick

File: pages/resources/quizzes/careerQuiz/index.tsx [hunk 2]
```diff
@@ -183,62 +176,106 @@ const CareerQuiz = () => {
...
-                quizViewState={quizViewState}
-              />
-            );
-          default:
-            return null;
-        }
-      })()}
-    </QuizTransition>
-  );
+  switch (stage) {
```
Comment (LuckyHariharan) [inline pages/resources/quizzes/careerQuiz/index.tsx:179]: simpler/clearer use of switch case

**Review by iamnotreddy — APPROVED** [review]
ran this locally and it looked pretty smooth - nice work on the refactor!

**Comment by LuckyHariharan** [comment]
> ran this locally and it looked pretty smooth - nice work on the refactor!

woohoo tysm!

File: pages/resources/quizzes/careerQuiz/index.tsx [hunk 3]
```diff
@@ -183,62 +176,108 @@ const CareerQuiz = () => {
...
+        >
+          {animationComplete && (
+            <CareerResults
+              onBackClick={handleBackClick}
+              quizViewState={quizViewState}
+            />
+          )}
+        </QuizTransition>
+      );
+    default:
```
Comment (vithushan19) [inline pages/resources/quizzes/careerQuiz/index.tsx:261]: when do you get to the default case?

**Review by vithushan19 — APPROVED** [review]
lgtm!

**Pull request merged**